coalib/bearlib/languages/Language.py

```diff
@@ -429,14 +471,105 @@ def limit_versions(language, limit, operator):
     else:
-
         versions = [version for version in language.versions
                     if operator(version, limit)]
     if not versions:
         raise ValueError('No versions left')
     return type(language)(*versions)
 
 
+def type_equals(language, comparable):
+    """
+    Checks if the given instances are of the same Language class and raises an
+    error otherwise.
+
+    :param language:
+        A `Language` instance
+    :param comparable:
+        Another `Language` instance
+    :return:
+        True
+    :raises TypeError:
+        If Language instances are not of the same class
+    """
+    if type(language) is type(comparable):
+        return True
+    else:
+        raise TypeError('Instances are not of the same class')
+
+
+def compare_versions(language, versions, operator):
+    """
+    Compares the version of a Language instance to the given set of versions
+    using gt, lt, ge or le operator. Comparison can be achieved if only the
+    given instance has a specified version. Otherwise, comparison of sets of
+    versions doesn't make sense and raises an error.
+
+    :param language:
+        A `Language` instance
+    :param versions:
+        A tuple of versions to be compared to instance's version.
+    :param operator:
+        The operator to use for the comparison.
+    :return:
+        True/False
+    :raises TypeError:
+        If Language instance has zero or more than one version
+    """
+    if len(language.versions) != 1:
+        raise TypeError('You have to specify ONE version')
+    else:
+        return all(operator(list(language.versions)[0], version)
+                   for version in versions)
+
+
+def compare(language, comparable, operator_):
+    """
+    Compares a Language instance depending on the type of comparable and the
+    given operator. In case comparable is not a Language instance, operators are
+    used for limiting the versions.
+
+    Comparison:
+    >>> Language['C'] == Language["C"]
+    True
+    >>> Language['py 2'] != Language['py 3']
+    True
+    >>> Language['Python 2.7'] < Language['Python 3.4']
+    True
+    >>> Language['Python 2.7'] > Language['py 3']
+    False
+    >>> Language['Python'] < Language['py 3']
+    Traceback (most recent call last):
+    ...
+    TypeError: You have to specify ONE version
+
+    Version Limiting:
+    >>> Language['python'] < 3
+    Python 2.7
+    >>> Language['python'] < 2
+    Traceback (most recent call last):
+    ...
+    ValueError: No versions left
+
+    :param language:
+        A `Language` instance.
+    :param comparable:
+        An instance to be compared to `language`.
+    :param operator:
+        The operator to use for the comparison.
+    :return:
+        True/False for same type comparison or a new `Language` instance with
+        limited versions.
+    """
+    if not isinstance(comparable, Language):
+        return limit_versions(language, comparable, operator_)
```

Comment:
I'm very unsure if we should really keep that version limiting by using comparison operator at all...